### ISSUE TITLE
faulty read

### ISSUE THREAD

**Issue by code4brot** [9elements/picoprog#20]
We've had several wrong reads using picoprog (+ osfc.io PCB enclosure).

This happened on reading several T540p chips. 
I think it's Macronix, will send the exact IC specificer(s) tomorrow when I have the boards under my fingers. 

Misbehavior took place in 2 forms: 
1. successful read, followup read -> "NO EEPROM", *dis- + reconnect IDC connector*, successful read, followup read -> "NO EEPROM", ...

2. 4M chip got read as 8M (with `-r` flag even 8M saved to disk) 

this happened on both `flashprog` and `flashrom`. 

Note on 1.) successful reads between fuckups seem consistent, sha1sum is equal.

Note on 2.)  The clips didn't move a bit. We only found that one by reverting to ch341a (after ordering new clips of course). 

Haven't tried any of this against the original pico-serprog firmware. Will try soon and report back!

**Comment by code4brot** [comment]
The chip was a MX25L3273E.

**Comment by code4brot** [comment]
Yep it does. No I'm not, at least not knowingly. I leave the other chip alone when reading the 4M chip.

**Comment by ArthurHeymans** [comment]
![images (6).jpeg](https://github.com/user-attachments/assets/357838de-f6f8-464e-a430-de038bde70f3)

If CS is not explicitly pulled high on the other chip it might be confused. Please try that.

**Comment by MDr164** [comment]
I found out that there is an issue on some chips with regards to how chunking is done right now. I got a potential fix but I don't have a chip here that actually shows that behavior. I'll push the update to the chunking this week, maybe give it another try after that.
